lib/widgets/bottom_sheet_menu.dart

```diff
@@ -10,16 +14,166 @@ class BottomSheetMenu extends StatefulWidget {
 }
 
 class _BottomSheetMenuState extends State<BottomSheetMenu> {
-  @override
-  void dispose() {
-    // dispose() akan dipanggil ketika widget tidak lagi dibutuhkan
-    super.dispose();
+  void _navigateToHome() {
+    Navigator.pop(context); // Tutup bottom sheet
+    Navigator.push(
+      context,
+      MaterialPageRoute(builder: (context) => const HomePage()),
+    );
+  }
+
+  void _navigateToMyProfile() {
+    Navigator.pop(context); // Tutup bottom sheet
+    Navigator.push(
+      context,
+      MaterialPageRoute(builder: (context) => const MyProfileScreen()),
+    );
+  }
+
+  void _logout() {
+    Navigator.pop(context); // Tutup bottom sheet
+    Navigator.push(
+      context,
+      MaterialPageRoute(builder: (context) => const LoginScreen()),
+    );
   }
 
   @override
   Widget build(BuildContext context) {
-    return const Center(
-      child: Text('showModalBottomSheet'),
+    double deviceHeight = MediaQuery.of(context).size.height;
+    double deviceWidth = MediaQuery.of(context).size.width;
+
+    Color purpleTextColor =
+        const Color.fromARGB(255, 181, 93, 190); // Format ARGB
+
+    return SizedBox(
+      height: deviceHeight / 3, // 1/3 dari tinggi layar
+      width: deviceWidth, // Lebar layar
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          GestureDetector(
+            onTap: _navigateToHome,
+            child: Padding(
+              padding: const EdgeInsets.only(left: 25.0),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.start,
+                children: [
+                  Padding(
+                    padding: const EdgeInsets.only(right: 22.0),
+                    child: SvgPicture.asset(
+                      'assets/images/icon_home.svg', // Sesuaikan path dengan lokasi ikon Home
+                      width: 24,
+                      height: 24,
+                      color: purpleTextColor,
+                    ),
+                  ),
+                  Text(
+                    'Home',
+                    style: TextStyle(
+                      fontSize: 20,
+                      color: purpleTextColor,
+                      fontWeight: FontWeight.bold,
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          ),
+          const SizedBox(height: 30), // Jarak antara 'Home' dan 'My Profile'
+          GestureDetector(
```

Comment:
setelah itu di looping dari setiap gesture detector sampai sizedbox..contohnya bisa dilihat menu cari loopingnya atau bisa dilihat disini https://stackoverflow.com/questions/56947046/flutter-for-loop-to-generate-list-of-widgets @eryagiandi09